src/workspace-data/data-table/shared/RenameColumnModal.ts

```diff
@@ -0,0 +1,125 @@
+import { Fragment, ReactNode, useState } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common';
+import { ValidatedInput } from 'src/components/input';
+import Modal from 'src/components/Modal';
+import { reportError } from 'src/libs/error';
+import { FormLabel } from 'src/libs/forms';
+import * as Utils from 'src/libs/utils';
+import validate from 'validate.js';
+
+export type RenameColumnModalProps = {
+  onDismiss: () => void;
+  onSuccess: () => void;
+  entityType: string;
+  attributeNames: string[];
+  oldAttributeName: string;
+  dataProvider: any;
+};
+
+export const RenameColumnModal = (props: RenameColumnModalProps): ReactNode => {
+  // State
+  const [newAttributeName, setNewAttributeName] = useState('');
+  const [isBusy, setIsBusy] = useState(false);
+  const { onDismiss, onSuccess, entityType, attributeNames, oldAttributeName, dataProvider } = props;
+
+  // Imposes same constraints on GCP and Azure:
+  // Composed of only letters, numbers, underscores, or dashes; regex match "[A-z0-9_-]+"
+  const mutualColumnNameErrors = validate.single(newAttributeName, {
+    presence: {
+      allowEmpty: false,
+      message: 'Column name is required',
+    },
+    format: {
+      pattern: '[A-Za-z0-9_-]+$',
+      flags: 'i',
+      message: 'Column name may only contain alphanumeric characters, underscores, and dashes.',
+    },
+    exclusion: {
+      within: attributeNames,
+      message: "'%{value}' already exists as an attribute name",
+    },
+  });
+
+  // On GCP only, cannot be one of these reserved words:
+  //   “name”
+  //   “entityType”
+  //   “${entityType}_id”, where ${entityType} is the name of the data table
+  const gcpColumnNameErrors =
+    dataProvider.providerName === 'Entity Service'
+      ? validate.single(newAttributeName, {
+          format: {
+            pattern: `^(?!name$|entityType$|${entityType}_id$).*`,
+            flags: 'i',
+            message: `Column name cannot be "name", "entityType" or "${entityType}_id".`,
+          },
+        })
+      : {};
+
+  // On Azure only, cannot begin with “sys_”
+  const azureColumnNameErrors =
+    dataProvider.providerName === 'WDS'
+      ? validate.single(newAttributeName, {
+          format: {
+            pattern: '^(?!sys_).*',
+            flags: 'i',
+            message: 'Column name cannot start with "sys_".',
+          },
+        })
+      : {};
+
+  const columnNameErrors = {
```

Comment:
@nawatts This concatenation leads the `disabled` to always be true since I guess it only checks it at first... is there a smart way to concatenate these or do I have to restructure the whole logic?

Reply:
Actually you know what I need to have completely different error checks per platform anyway so I guess i don't need to concatenate them

Reply:
`validate.single` returns an array of messages or `undefined`. This isn't concatenating those arrays though, it's creating an object with keys "mutualColumnNameErrors", etc.

They can be concatenated like:
```ts
const columnNameErrors = [
  ...(mutualColumnNameErrors || []),
  ...(gcpColumnNameErrors || []),
  ...(azureColumnNameErrors || []),
];
```

Reply:
Ugh that still doesn't get disabled to work correctly.  Will have to rethink this on Monday.

Reply:
Oh, disabled is currently based on converting `columnNameErrors` to a boolean, but an array is always "truthy".
```js
disabled: isBusy || !!columnNameErrors,
```

It would need to be changed to check the length of `columnNameErrors`.
```js
disabled: isBusy || columnNameErrors.length > 0,
```

Reply:
Thank you!